pkg/scaling/scale_handler.go

```diff
@@ -566,9 +566,19 @@ func (h *scaleHandler) getScaledObjectState(ctx context.Context, scaledObject *k
 		}
 
 		for _, spec := range metricSpecs {
-			// skip cpu/memory resource scaler, these scalers are also always Active
+			// if cpu/memory resource scaler has minReplicas==0 & at least one external
+			// trigger exists -> object can be scaled to zero
+			//
 			if spec.External == nil {
-				isScaledObjectActive = true
+				cpuMemCount := 0
+				for _, trigger := range scaledObject.Spec.Triggers {
+					if trigger.Type == "cpu" || trigger.Type == "memory" {
+						cpuMemCount++
+					}
+				}
```

Comment:
Shouldn't we calculate this before the loop? I mean, if we have 3 external metrics we will evaluate the `cpuMemCount` value 3 times

Reply:
its in the `if spec.External == nil` branch so whenever its non external trigger the `for` gets triggered. but if we have multiple cpu/mem triggers it would evaluate multiple times, thats true. I can move it to the beginning of the function which would mean it would evaluate even if there were no cpu/mem triggers. 
What you want me to do?

Reply:
I guess that it's better evaluating it once always than X amount of times in edge cases. 
Moving it before it's O(n) and inside the loop is O(n2)